lib/pbench/cli/agent/commands/triggers/list.py

```diff
@@ -0,0 +1,83 @@
+"""
+pbench-list-triggers
+
+This script will list all tool triggers from all tool groups, or list tool
+triggers from a specific group.
+
+"""
+import logging
+import sys
+
+import click
+
+from pbench.agent.utils import setup_logging
+from pbench.cli.agent import CliContext, pass_cli_context
+from pbench.cli.agent.commands.triggers.base import TriggerCommand
+from pbench.cli.agent.options import common_options
+
+LOG = logging.getLogger(__name__)
+
+
+class TriggerList(TriggerCommand):
+    """List register triggers"""
+
+    def __init__(self, context):
+        super().__init__(context)
+
+        setup_logging(debug=False, logfile=None)
+
+    def execute(self):
+        if not self.pbench_run.exists():
+            # Silently exit if we don't have a pbench run directory.
+            return 1
+
+        if not self.context.group:
+            # list tool triggers in all groups
+            if len(self.groups) == 0:
+                LOG.error("%s: error fetching list of tool groups", self.name)
+                return 1
+
+            for group in self.groups:
+                trigger = self.gen_tools_group_dir(group) / "__trigger__"
+                if trigger.exists():
+                    with open(trigger, "r") as f:
+                        data = f.read()
+                        print("%s: %s" % (group, data.strip()))
+        else:
+            # list tool triggers in all groups
+            if self.verify_tool_group(self.context.group) != 0:
+                LOG.error(
+                    '%s: bad tool group specified, "%s"\n',
+                    self.name,
+                    self.context.group,
+                )
+                return 1
+            trigger = self.tool_group_dir / "__trigger__"
+            if trigger.exists():
+                with open(trigger) as f:
+                    data = f.read()
+                    print(data.strip())
```

Comment:
shouldn't this be `log` instead of `print`?

Reply:
This is the output of the list command; so, no. ;-)